File: engine/engine-model/src/main/scala/tools/packet/family/FamilyJoinResponse.scala
package tools.packet.family

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class FamilyJoinResponse(private var _accepted: Boolean, private var _characterNameAdded: String) extends PacketInput {
  def accepted: Boolean = _accepted

  def characterNameAdded: String = _characterNameAdded

  override def opcode(): SendOpcode = SendOpcode.FAMILY_JOIN_REQUEST_RESULT
}

File: engine/engine-model/src/main/java/tools/packet/family/FamilyJoinResponse.java
package tools.packet.family;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record FamilyJoinResponse(Boolean accepted, String characterNameAdded) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FAMILY_JOIN_REQUEST_RESULT;
   }
}